### PR TITLE
[SIM115] Allow `open` followed by `close`

### PR DESCRIPTION
**Pull request by harupy**
## Summary

https://github.com/astral-sh/ruff/discussions/7907

## Test Plan

New test cases

### CODE REVIEW

File: crates/ruff_linter/resources/test/fixtures/flake8_simplify/SIM115.py [hunk 1]
```diff
@@ -42,3 +42,7 @@
 with contextlib.ExitStack() as exit_stack:
     exit_stack_ = exit_stack
     f = exit_stack_.enter_context(open("filename"))
+
+# OK
+open("filename").close()
```
[the remark this answers is not in the document]
Comment (harupy) [reply crates/ruff_linter/resources/test/fixtures/flake8_simplify/SIM115.py:47]:
```suggestion
# OK (file is closed immdeidately)
open("filename").close()
```

comment like this?

[the remark this answers is not in the document]
Comment (konstin) [reply crates/ruff_linter/resources/test/fixtures/flake8_simplify/SIM115.py:47]: More like "Used to clear an existing file (https://stackoverflow.com/a/2769090/3549270)"

There's `Path.touch` now but i'm surprised there's nothing similar from clearing a file.

**Comment by konstin** [comment]
For append instead of write cases, e.g. https://github.com/apache/airflow/blob/e239dcf644b6e68896923d84dd49e9cbb6271373/airflow/utils/log/file_processor_handler.py#L150, i'd recommend using `Path.touch` instead, but this might better be a separate rule.

File: crates/ruff_linter/resources/test/fixtures/flake8_simplify/SIM115.py [hunk 2]
```diff
@@ -42,3 +42,7 @@
 with contextlib.ExitStack() as exit_stack:
     exit_stack_ = exit_stack
     f = exit_stack_.enter_context(open("filename"))
+
+# OK (quick one-liner clear file contents)
```
Comment (harupy) [inline crates/ruff_linter/resources/test/fixtures/flake8_simplify/SIM115.py:46]:
```suggestion
# OK (quick one-liner to clear file contents)
```

**Pull request merged**